docs/sources/developers/plugins/add-support-for-annotations.md

```diff
@@ -8,140 +8,26 @@ This guide explains how to add support for [annotations]({{< relref "../../dashb
 
 ## Add annotations support to your data source
 
-To add support for annotations to an existing data source, you need to:
-
-- Enable annotations support
-- Override the `annotationQuery` method
-- Construct annotation events
-
-### Enable annotations support
-
-Tell Grafana that your data source plugin can return annotations events by adding `"annotations": true` to the [plugin.json]({{< relref "metadata.md" >}}) file.
-
-```json
-{
-  "annotations": true
-}
-```
-
-### Override the `annotationQuery` method
-
-In `DataSource.ts`, override the `annotationQuery` method from `DataSourceApi`.
-
-```ts
-async annotationQuery(options: AnnotationQueryRequest<MyQuery>): Promise<AnnotationEvent[]> {
-  return [];
-}
-```
-
-### Construct annotation events
-
-Return an array of [AnnotationEvent]({{< relref "../../packages_api/data/annotationevent.md" >}}).
-
-```ts
-async annotationQuery(options: AnnotationQueryRequest<MyQuery>): Promise<AnnotationEvent[]> {
-  const events: AnnotationEvent[] = [];
-
-  const date = new Date();
-
-  const event: AnnotationEvent = {
-    time: date.valueOf(),
-    text: 'foo',
-    tags: ['bar'],
-  };
-
-  events.push(event);
-
-  return events;
-}
-```
-
-## Region annotations
-
-[Region annotations]({{< relref "../../dashboards/annotations.md#adding-regions-events" >}}) have a start and end time. This can for example be used to annotate maintenance windows or downtime.
-
-To return a region annotation, set the `timeEnd`, and `isRegion` properties.
+To enable annotation support for your data source, you need to add the following two lines of code. Grafana uses your default query editor for editing annotation queries.
 
-```ts
-const regionEvent: AnnotationEvent = {
-  time: startDate.valueOf(),
-  timeEnd: endDate.valueOf(),
-  isRegion: true,
-  text: 'foo',
-  tags: ['bar'],
-};
-```
+1. Add `"annotations": true` to the [plugin.json]({{< relref "metadata.md" >}}) file.
 
-## Build a annotation query editor
+   **plugin.json**
 
-Let users write custom annotation queries to only display the annotation events they care about, by adding a _query editor_. You only need to build a query editor if you want to let users query or filter annotations.
-
-> **Note:** Annotation query editors have yet to receive support for React. The instructions here are given for Angular. Fortunately, you can run Angular even in a plugin otherwise written using React. This section will be updated once React support for annotation queries editors is available.
-
-1. Create a file called `AnnotationQueryEditor.ts` in the `src` directory, with the following content.
-
-   ```ts
-   export class AnnotationQueryEditor {
-     static templateUrl = 'partials/annotations.editor.html';
-
-     annotation: any;
-
-     constructor() {
-       this.annotation.queryText = this.annotation.queryText || '';
-     }
+   ```json
+   {
+     "annotations": true
```

Comment:
is this still required?  I would love to get rid of as much of this as possible!

Reply:
I tried removing it, but the data source wouldn't even show up in the drop-down, so it seems like it's still required.